client/src/templates/Challenges/classic/show.tsx

```diff
@@ -502,7 +502,12 @@ class ShowClassic extends Component<ShowClassicProps, ShowClassicState> {
         usesMultifileEditor={usesMultifileEditor}
       >
         <LearnLayout>
-          <Helmet title={windowTitle} />
+          <Helmet title={windowTitle}>
+            <style>
+              {'.page-wrapper{height: 100%;}' +
+                '.default-layout, #learn-app-wrapper, #content-start, #content-start > div {display: flex;flex-direction: column;min-height: 0;flex: 1 1 auto;}'}
+            </style>
```

Comment:
It seems that most styles can be moved to related files (`global.css`, `classic.css`, and `learn.css`). However, if we move `.page-wrapper{height: 100%;}` to related css file (`global.css`) in the same way and import it, the display of the landing page and learn page will collapse because `.page-wrapper` is flex container, so `.page-wrapper{height: 100%;}` needs to be overridden only when the `ShowClassic` component is mounted. To do that, we need to leave the line alone in the `ShowClassic` component's Helmet.(It may not be best practice, but so far I don't have another idea to do so.)

Reply:
Removing the min-height: 0 from the .default-layout seems to be resolving this page collapse issue.

Reply:
Actually, removing `the min-height : 0` from `.default-layout` after moving `.page-wrapper{height: 100%;}` to `global.css` seems to make the desktop-layout and mobile-layout of Challenges and Steps not fully vertically flexible and when camper reduces the height of browser, vertical scrollbar of `body` will appear like below gif.

![removed_min_height_zero ](https://user-images.githubusercontent.com/44451585/201077999-6ea87403-337f-4be1-b69e-be344917784e.gif)

Reply:
When `.desktop-layout` or `.mobile-layout` is displayed, `.default-layout` is only child of `.page-wrapper`.
So `min-height: 0;` was deleted from the `.default-layout` after moving `.page-wrapper{height: 100%;}` to `global.css`, while by using `.default-layout:only-child { min-height: 0; }`, the code avoid appearing vertical scrollbar of `body` in desktop and mobile layout.